ruy/mul_params.h

```diff
@@ -59,9 +70,14 @@ class MulParams final {
   // The bias vector data, if not null.
   const AccumScalar* bias() const { return storage_.bias; }
   void set_bias(const AccumScalar* ptr) { storage_.bias = ptr; }
-  // Only for non-floating-point cases. The fixed-point part (i.e. the mantissa)
-  // of the multiplier by which accumulators are multiplied before being casted
-  // to the destination type.
+  // Only for non-floating-point cases. The fixed-point part  of the multiplier
+  // by which accumulators are multiplied before being casted to the destination
+  // type. This is a fixed-point quantity with 0 integer bits. Since
+  // (as explained in the class comment) AccumScalar must be std::int32_t,
+  // that means that the fixed-point format is Q0.31. For example,
+  // a multiplier_fixedpoint value of 2^30 has the effect of multiplying
+  // by one half (1/2). More generally, the effect is to multiply by
+  // (multiplier_fixedpoint / (2^31)).
```

Comment:
Mention what the exponent does here. I think, in general it's

(multiplier_fixedpoint << multiplier_exponent) >> 31 ?

Reply:
ah ok i'll update the PR. it's a little more subtle because the exponent could be positive or negative.
the idea is:
1. left shift by positive part of exponent.
2. perform the fixed-point multiplication
3. right shift by the negative part of exponent, with suitable rounding.

The reference implementation is: https://github.com/google/ruy/blob/master/ruy/apply_multiplier.cc#L74-L83

Reply:
I feel like it's fair to hand wave `<< -x` a bit to get the idea across

Reply:
updated the PR with lots more comments here.